src/test-run/index.js

```diff
@@ -228,5 +240,5 @@ ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
         }
     }
 
-    return new Promise((resolve, reject) => this.pendingRequest = { resolve, reject });
+    return new Promise((resolve, reject) => this.pendingRequest = { statusId: driverStatus.id, resolve, reject });
```

Comment:
I've had a hard time understanding this workflow. As I see it, we should split handler into transport layer and execution layer:
```js
 ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
  // 1. Check status id and return cached response here
  // 2. Get response by calling `handleDriverRequest`
  // 3. Cache response and send it back
};
```

So, business logic will be isolated from transport-related stuff.